File: android/app/src/main/java/com/pdftoolkit/app/data/model/ToolType.kt
package com.pdftoolkit.app.data.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.CallSplit
import androidx.compose.material.icons.outlined.Compress
import androidx.compose.material.icons.outlined.Collections
import androidx.compose.material.icons.outlined.Difference
import androidx.compose.material.icons.outlined.Image
import androidx.compose.ui.graphics.vector.ImageVector

enum class ToolType(
    val title: String,
    val description: String,
    val route: String
) {
    IMAGE_TO_PDF(
        title = "Image to PDF",
        description = "Convert photos & gallery images into a clean PDF document",
        route = "image_to_pdf"
    ),
    MERGE_PDF(
        title = "Merge PDFs",
        description = "Combine 2 or more PDF documents into a single file",
        route = "merge_pdf"
    ),
    SPLIT_PDF(
        title = "Split PDF",
        description = "Extract selected pages or custom page ranges into a new PDF",
        route = "split_pdf"
    ),
    COMPRESS_PDF(
        title = "Compress PDF",
        description = "Shrink PDF file size while keeping visual clarity crisp",
        route = "compress_pdf"
    ),
    PDF_TO_IMAGE(
        title = "PDF to Images",
        description = "Export individual PDF pages as high quality JPG or PNG",
        route = "pdf_to_image"
    );

    val icon: ImageVector
        get() = when (this) {
            IMAGE_TO_PDF -> Icons.Outlined.Image
            MERGE_PDF -> Icons.Outlined.Difference
            SPLIT_PDF -> Icons.Outlined.CallSplit
            COMPRESS_PDF -> Icons.Outlined.Compress
            PDF_TO_IMAGE -> Icons.Outlined.Collections
        }
}

File: ios/PDFToolkit/Models/ToolType.swift
import SwiftUI

enum ToolType: String, CaseIterable, Identifiable, Codable {
    case imageToPdf = "image_to_pdf"
    case mergePdf = "merge_pdf"
    case splitPdf = "split_pdf"
    case compressPdf = "compress_pdf"
    case pdfToImage = "pdf_to_image"

    var id: String { rawValue }

    var title: String {
        switch self {
        case .imageToPdf: return "Image to PDF"
        case .mergePdf: return "Merge PDFs"
        case .splitPdf: return "Split PDF"
        case .compressPdf: return "Compress PDF"
        case .pdfToImage: return "PDF to Images"
        }
    }

    var description: String {
        switch self {
        case .imageToPdf: return "Convert photos & gallery images into a clean PDF"
        case .mergePdf: return "Combine 2 or more PDF documents into a single file"
        case .splitPdf: return "Extract selected pages or custom ranges into a new PDF"
        case .compressPdf: return "Shrink PDF file size while keeping visual clarity crisp"
        case .pdfToImage: return "Export individual PDF pages as high quality JPG or PNG"
        }
    }

    var iconName: String {
        switch self {
        case .imageToPdf: return "photo.on.rectangle.angled"
        case .mergePdf: return "doc.on.doc"
        case .splitPdf: return "arrow.triangle.branch"
        case .compressPdf: return "arrow.down.right.and.arrow.up.left"
        case .pdfToImage: return "photo.stack"
        }
    }
}
